### PR TITLE
feat(database, selection): improve DnD for database block

### PR DESCRIPTION
**Pull request by fundon**
* [x] drag and drop a block into empty database
* [x] improve DragIndicator position
* [x] improve DragHandle position
* [x] BlockHub 

https://user-images.githubusercontent.com/27926/226604094-2ce7d9f6-eebb-45be-b0b7-d49d59211604.mov

### CODE REVIEW

**Comment by fundon** [comment]
> One thing I want to confirm is whether the following two visual effects need to be consistent?

If the database is empty, the width of drag handle line is equals database block' width, otherwise it equals to the specific block' width.

**Comment by fundon** [comment]
@zqran 


https://user-images.githubusercontent.com/27926/229326372-21b601e1-3546-4428-a2e4-0fb76ed76dda.mov

File: packages/blocks/src/components/block-hub.ts [hunk 1]
```diff
@@ -727,16 +726,40 @@ export class BlockHub extends NonShadowLitElement {
...
+            ?.getBoundingClientRect();
+          assertExists(headerBounds);
+          targetRect = new DOMRect(
+            headerBounds.left,
+            headerBounds.bottom + 1,
+            targetRect.width,
+            1
+          );
+        }
+      }
```
Comment (lawvs) [inline packages/blocks/src/components/block-hub.ts:751]: I suggest adding some comments about drag details here for better understanding

**Review by lawvs — CHANGES_REQUESTED** [review]

**Comment by zqran** [comment]
> @zqran Throws an exception when all rows are moved out of the database

Fixed.

**Review by doodlewind — APPROVED** [review]

**Pull request merged**